README.md

```diff
@@ -1,526 +1,917 @@
 # Decorators
 
-This README describes a new proposal, to be presented to TC39 in the September 2020 meeting.
+Decorators are a proposal for extending JavaScript classes which is widely adopted among developers in transpiler environments, with broad interest in standardization. TC39 has been iterating on decorators proposals for over five years. This document describes a new proposal for decorators based on elements from all past proposals.
 
-# Introduction
+This README describes the current decorators proposal, which is a work in progress. For previous iterations of this proposal, see the commit history of this repository.
 
-Decorators are a proposal for extending JavaScript classes which is widely adopted among developers in transpiler environments, with broad interest in standardization. TC39 has been iterating on decorators proposals for over five years. This document describes a new proposal for decorators based on elements from all past proposals.
+## Introduction
 
-**Decorators** `@decorator` are *functions* called on class elements or other JavaScript syntax forms during definition, potentially *wrapping* or *replacing* them with a new value returned by the decorator.
+**Decorators** are *functions* called on classes, class elements, or other JavaScript syntax forms during definition.
 
-A decorated class field is treated as wrapping a getter/setter pair for accessing that storage. Decorated storage is useful for observation/tracking, which has been a pain point for the original legacy/experimental decorators combined with [[Define]] semantics for class fields. These semantics are based on Michel Weststrate's ["trapping decorators" proposal](https://github.com/tc39/proposal-decorators/issues/299).
+```js
+@defineElement("my-class")
+class C extends HTMLElement {
+  @reactive prop clicked = false;
+}
+```
 
-Decorators may also annotate a class element with *metadata*. These are simple, unrestricted object properties, which are collected from all decorators which add them, and made available as a set of nested objects in the `[Symbol.metadata]` property.
+Decorators have three primary capabilities:
 
-By making decorators always simply wrap what they are decorating, rather than performing other transformations, this proposal aims to meet the following requirements:
-- The class "shape" is visible without executing the code, making decorators more optimizable for engines.
-- Implementations can work fully on a per-file basis, with no need for cross-file knowledge.
-- No new namespace or type of second-class value is added--decorators are functions.
+1. They can **replace** the value that is being decorated with a _matching_ value that has the same semantics. (e.g. a decorator can replace a method with another method, a field with another field, a class with another class, and so on).
```

Comment:
this may be answered elsewhere, but will this be validated? in other words, must a class replacement be a `class` constructor, must a method be another method (`[[HomeObject]]` and `.prototype` aside, altho that raises a question about the resulting method’s home object and .prototype)?

Reply:
The validation is mentioned later on, so far all it would be is that the return value is a particular type (e.g. `function` for methods, has `[[Construct]]` for classes). We could try to add additional constraints, but I believe in #334 the conclusion was that there weren't too many constraints we could add. If you think there are additional constraints that should be added, we can discuss in the next meeting (or you can open up an issue/pr with suggestions)